core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -66,8 +67,16 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
                                     contentType.subtype().endsWith("+json"))) {
             final ObjectReader reader = readers.computeIfAbsent(expectedResultType, mapper::readerFor);
             if (reader != null) {
-                return reader.readValue(request.content().toString(
-                        contentType.charset().orElse(StandardCharsets.UTF_8)));
+                try {
+                    return reader.readValue(request.content().toString(
+                            contentType.charset().orElse(StandardCharsets.UTF_8)));
+                } catch (JsonProcessingException e) {
+                    // Do not create the Exception with cause and throw IllegalArgumentException
```

Comment:
Out of curiosity, why not add the cause? It might provide useful information when using `LoggingService`.

Reply:
The reason is that If I add the cause, the `DefaultExceptionHandler` cannot handle this exception because the `handleException(ctx, req, rootCause)` is called with the `rootCause`.
However, I talk to @trustin and we have reached the conclusion that we change the part that passing the `rootCause`.
I will update PR and please review again after that. :-)